app/helpers/assessment_query_helper.rb

```diff
@@ -69,35 +69,77 @@ def paginate(assessments, entries, page)
     assessments.each do |assessment|
-      reported_condition = assessment.reported_condition
       details = {
         id: assessment[:id],
-        symptomatic: assessment.symptomatic ? 'Yes' : 'No',
-        who_reported: assessment.who_reported,
-        created_at: assessment.created_at,
-        threshold_condition_hash: reported_condition&.threshold_condition&.threshold_condition_hash,
-        symptoms: reported_condition&.symptoms&.uniq(&:name)
+        symptomatic: assessment[:symptomatic] ? 'Yes' : 'No',
+        who_reported: assessment[:who_reported],
+        created_at: assessment[:created_at],
+        passes_threshold_data: {}
       }
 
-      passes_threshold_data = {}
-      symptoms&.pluck(:name)&.each do |symptom_name|
-        symptom = assessment.get_reported_symptom_by_name(symptom_name)
-        value = symptom&.value
-        # NOTE: We must check if the value is nil here before making this change.
-        # Otherwise, text responses of "Yes" will show "No" in each row because the value for each symptom is still empty, for example.
-        if symptom&.type == 'BoolSymptom' && !value.nil?
-          value = value == true ? 'Yes' : 'No'
+      reported_condition = reported_conditions[reported_condition_ids_by_assessment_id[assessment[:id]]]
+      unless reported_condition.nil?
+        details[:threshold_condition_hash] = reported_condition[:hash]
+        details[:symptoms] = reported_condition[:symptoms]
+        details[:symptoms]&.each do |symptom|
+          value = symptom.value
+          # NOTE: We must check if the value is nil here before making this change.
+          # Otherwise, text responses of "Yes" will show "No" in each row because the value for each symptom is still empty, for example.
+          value = value == true ? 'Yes' : 'No' if symptom[:type] == 'BoolSymptom' && !value.nil?
+          details[symptom[:name]] = value.nil? ? '' : value
+
+          # Determine if reported symptom passes threshold based on threshold symptom by threshold_condition_hash and name
+          threshold_symptom = threshold_symptoms[reported_condition[:hash]][:symptoms][symptom[:name]]
+          details[:passes_threshold_data][symptom[:name]] = assessment.symptom_passes_threshold(symptom, threshold_symptom)
         end
-        details[symptom_name] = value.nil? ? '' : value
-        passes_threshold_data[symptom_name] = assessment.symptom_passes_threshold(symptom_name)
       end
-      details[:passes_threshold_data] = passes_threshold_data
+
       table_data << details
     end
-    { table_data: table_data, symptoms: symptoms || [], total: assessments.total_entries }
+
+    { table_data: table_data, symptoms: threshold_symptoms.values.flat_map { |s| s[:symptoms]&.values }.uniq(&:name) || [], total: assessments.total_entries }
+  end
+
+  # Gets all unique symptoms (based on name and threshold_condition_hash) for a given array of assessment IDs.
+  def get_unique_threshold_symptoms(assessment_ids)
```

Comment:
where is this method used?

Reply:
ah, this was originally renamed from `get_unique_symptoms_for_assessments` but is no longer being used, so I will remove it

Reply:
Ok feel free to resolve this once you push that change